cql-engine/src/main/java/org/opencds/cqf/cql/elm/execution/IntersectEvaluator.java

```diff
@@ -65,8 +66,28 @@ public static Object intersect(Object left, Object right)
                 return null;
             }
 
-            Object max = GreaterEvaluator.greater(leftStart, rightStart) ? leftStart : rightStart;
-            Object min = LessEvaluator.less(leftEnd, rightEnd) ? leftEnd : rightEnd;
+            Boolean leftStartGtRightStart = GreaterEvaluator.greater(leftStart, rightStart);
+            Boolean leftEndLtRightEnd = LessEvaluator.less(leftEnd, rightEnd);
+
+            Object max;
+            if (leftStartGtRightStart == null && precision != null)
+            {
+                max = ((BaseTemporal) leftStart).getPrecision().toString().equals(precision) ? leftStart : rightStart;
+            }
+            else
+            {
+                max = leftStartGtRightStart == null ? null : leftStartGtRightStart ? leftStart : rightStart;
+            }
+
+            Object min;
+            if (leftEndLtRightEnd == null && precision != null)
+            {
+                min = ((BaseTemporal) leftEnd).getPrecision().toString().equals(precision) ? leftEnd : rightEnd;
+            }
+            else
+            {
+                min = leftEndLtRightEnd == null ? null : leftEndLtRightEnd ? leftEnd : rightEnd;
+            }
 
             return new Interval(max, true, min, true);
```

Comment:
The interval returned needs to have exclusive boundaries if the max and min are null

Reply:
I should say, the start boundary should be exclusive if the max is null, and the end boundary should be exclusive if the min is null.